feat(ui): Add new session category to the alert wizard

matejminar opened the pull request:
![image](https://user-images.githubusercontent.com/9060071/133258526-c37cee27-06de-425f-a829-24bdd0915f80.png)

Adding `Session` category to the alert wizard with two brand new options: `Crash Free Session Rate` and `Crash Free User Rate`.

It does not do anything yet, it's a feature flagged work in progress.

matejminar committed:
feat(ui): Add new session category to the alert wizard

billyvg commented on static/app/views/alerts/wizard/index.tsx:
```diff
@@ -180,19 +180,21 @@ class AlertWizard extends Component<Props, State> {
-                    />
-                  </OptionsWrapper>
-                ))}
+                {alertWizardCategories(organization).map(
```
What about calling this `getAlertWizardCategories`?

matejminar replied:
sounds good, let me change that 👍

billyvg commented on static/app/views/alerts/wizard/options.tsx:
```diff
@@ -40,33 +45,44 @@ export const AlertWizardAlertNames: Record<AlertType, string> = {
+    },
+  ];
+
+  if (org.features.includes('crash-rate-alerts')) {
+    options.splice(1, 0, {
+      categoryHeading: t('Sessions'),
+      options: ['crash_free_sessions', 'crash_free_users'],
+    });
+  }
```
Would this be easier to read if it was part of the `options` declaration? That way you'd be able to see the intended ordering instead of having to parse the `splice()` call.

matejminar replied:
I agree, splice is never easy to parse.

matejminar committed:
rename alertWizardCategories, refactor away splice

matejminar committed:
Merge branch 'master' into feat/session-alert-categories

loewenheim committed:
fix: Type error (#28628)

Commit:
feat(symbolication): Add symbolication status to project details (NATIVE-209) (#28593)

This adds a field in the project details which exposes the health of
the project's symbolication processing.  The aim is to let the UI show
warnings on the project if we are experiencing processing problems.

Currently returns dummy data, but a followup will change this to read
from 3 sentry options to return the correct value.

NATIVE-209

loewenheim committed:
feat: Add option for low priority metrics submission rate [NATIVE-185] (#28629)

Zylphrex committed:
chore(performance): Remove legacy key transactions endpoint (#28383)

Follow up to #28382 to remove the endpoints as they won't be in use anymore.

shruthilayaj committed:
feat(chart-unfurl): Add multi y-axis for chart unfurl (#28604)

Adding the ability to unfurl multi y-axis charts.

PixelDust22 committed:
feat(growth): Add default business for all experiment (#28633)

* add default business for all experiment

* change name

wedamija committed:
fix(semver): Fix autocomplete tag functionality to use string prefixes rather than semver cols (#28622)

Previously a search like `1.2` would be converted internally to `1.2.*`. This excludes any versions
like `1.20`, `1.200`, etc. To make this work as the user expects, instead we now just use string
prefixing. This should work in most cases, but if there are versions with 0 prefixes then those will
need to be included in the search string for it to work. So for example, `1.020.0` won't show up if
someone types `1.2`, only if they type `1.`, `1.0`, etc.

We still sort these results using semver sort, so they're returned in the order that the user
expects.

k-fish committed:
ref(ui): Add performance event view hook (#28572)

ref(ui): Add performance event view hook

This adds a hook to avoid having to constantly drill event view, especially for our generic discover query specializations. Essentially if any specialization now uses the 'DiscoverQueryPropsWithContext' type you no longer have to provide eventView and orgSlug, which should reduce a lot of boilerplate throughout performance. It's not on by default since there is likely some undefined checks and tests that might need to be rewritten (to use a provider) if you choose to use this.

Other:
- Giving a context wrapper to always check if a context is defined, we'll see if it's helpful, and if so we can raise this up to be used everywhere in the UI or talk about it during TSC.

maxiuyuan committed:
audit log related stuff for project quotas (#28620)

joshuarli committed:
chore: upgrade grpcio (#28623)

markstory committed:
fix(deletions) Handle organizations with orphaned commits (#28636)

We have a few organization that have 0 repositories, but still have
commits. Make CommitAuthor deletions record based so we can also remove
commits that would block a commit author deletion.

Fixes SENTRY-S5Y

markstory committed:
ref(deletion) Consolidate group deletion and improve logging (#28475)

We had two different implementations of group deletion. This change
consolidates those into a single path and fixes split group hashes being
removed by list page delete operations.

I've also added the organization_id to group deletion messages. This
helps improve log queries like 'did acme org delete any issues recently'
which comes up when customers notice discrepancies between the events
they currently have and the usage we've tracked.

davidenwang committed:
ref(select): Choices -> options AppStoreConnect (#28616)

Changes from legacy react-select choices to options in step 2 and step 5 of this modal

AniketDas-Tekky committed:
feat(alert-rule): Created an alert rule mediator (#28639)

New mediator handles the execution of webhooks based on an alert rule triggering.
Bonus: Added some logging for easier debugging in the future

Fixes API-2075

evanpurkhiser committed:
ref(teams): Update teamSelect to use useTeams hook (#28615)

davidenwang committed:
feat(store): Projects util now updates render-props projects when store emits updates (#28610)

Currently, the <Projects> util component does not react to updates from the ProjectsStore. This means there can be instances when project data has changed, but the application doesn't seem to reflect it. One such example is creating an issue alert or metric alert. Adding a team to a project should affect the state of both of these components as adding a team to a project in one dropdown should enable it in the other dropdown.

priscilawebdev commented on static/app/views/alerts/wizard/index.tsx:
```diff
@@ -180,19 +180,21 @@ class AlertWizard extends Component<Props, State> {
+                      {i > 0 && <Styledh2>{categoryHeading}</Styledh2>}
+                      <RadioPanelGroup
+                        choices={options.map(alertType => {
+                          return [alertType, AlertWizardAlertNames[alertType]];
```
nit: should this const have the first letter lower case? 
```suggestion
                          return [alertType, alertWizardAlertNames[alertType]];
```

matejminar replied:
there are lots of constants like this in that file, I didn't want to go on a refactoring spree because this PR would get messy

priscilawebdev commented on static/app/views/alerts/wizard/options.tsx:
```diff
@@ -181,6 +194,24 @@ export const AlertWizardPanelContent: Record<AlertType, PanelContent> = {
   },
+  crash_free_sessions: {
+    description: t(
+      'A session begins when a user starts the application and ends when it’s closed or sent to the background.  A crash is when a session ends due to an error and this type of alert lets you monitor when those crashed sessions exceeds a threshold. This lets you get a better picture of the health of your app.'
```
```suggestion
      'A session begins when a user starts the application and ends when it’s closed or sent to the background. A crash is when a session ends due to an error and this type of alert lets you monitor when those crashed sessions exceed a threshold. This lets you get a better picture of the health of your app.'
```

priscilawebdev commented on static/app/views/alerts/wizard/options.tsx:
```diff
@@ -181,6 +194,24 @@ export const AlertWizardPanelContent: Record<AlertType, PanelContent> = {
+  },
+  crash_free_users: {
+    description: t(
+      'Crash Free Users is the percentage of distinct users that haven’t experienced a crash and so this type of alert tells you when the overall user experience dips below a certain unaccpetable threshold.'
```
```suggestion
      'Crash Free Users is the percentage of distinct users that haven’t experienced a crash and so this type of alert tells you when the overall user experience dips below a certain unacceptable threshold.'
```

matejminar replied:
Thanks for catching that typo!

priscilawebdev commented on static/app/views/alerts/wizard/index.tsx:
```diff
@@ -180,19 +180,21 @@ class AlertWizard extends Component<Props, State> {
+                {getAlertWizardCategories(organization).map(
+                  ({categoryHeading, options}, i) => (
+                    <OptionsWrapper key={categoryHeading}>
+                      {i > 0 && <Styledh2>{categoryHeading}</Styledh2>}
```
nit pick: 
```suggestion
                      {i > 0 && <StyledH2>{categoryHeading}</StyledH2>}
```

matejminar replied:
If we want to refactor this, I'll change it to `CategoryTitle` so that it's more descriptive. 👍

matejminar committed:
ref: rename Styledh2 to CategoryTitle

matejminar committed:
Apply suggestions from code review

Co-authored-by: Priscila Oliveira <priscilawebdev@gmail.com>

matejminar committed:
Merge branch 'master' into feat/session-alert-categories
